File: src/main/java/com/timeback/ui/domain/model/ActionResult.java
package com.timeback.ui.domain.model;

import java.util.Map;

/**
 * CT-04 §3.3 사용자 작업 결과
 */
public abstract class ActionResult {

    private ActionResult() {}

    public static final class Success extends ActionResult {}

    public static final class Blocked extends ActionResult {
        private final BlockReason reason;
        public Blocked(BlockReason reason) { this.reason = reason; }
        public BlockReason getReason() { return reason; }
    }

    public static final class RetryableFailure extends ActionResult {
        private final ErrorReason reason;
        public RetryableFailure(ErrorReason reason) { this.reason = reason; }
        public ErrorReason getReason() { return reason; }
    }

    public static final class PartialFailure extends ActionResult {
        private final Map<String, Boolean> details;
        public PartialFailure(Map<String, Boolean> details) { this.details = details; }
        public Map<String, Boolean> getDetails() { return details; }
    }

    public static final class Failure extends ActionResult {
        private final ErrorReason reason;
        public Failure(ErrorReason reason) { this.reason = reason; }
        public ErrorReason getReason() { return reason; }
    }
}

File: src/main/java/com/timeback/ui/domain/model/ActionResult.kt
package com.timeback.ui.domain.model

/**
 * CT-04 §3.3 사용자 작업 결과 — frontend-components.md 매핑
 */

sealed interface ActionResult {
    data object Success : ActionResult
    data class Blocked(val reason: BlockReason) : ActionResult
    data class RetryableFailure(val reason: ErrorReason) : ActionResult
    data class PartialFailure(val details: Map<String, Boolean>) : ActionResult
    data class Failure(val reason: ErrorReason) : ActionResult
}
